js/index.js

```diff
@@ -0,0 +1,91 @@
+const previousButton = document.getElementById("previous");
+const nextButton = document.getElementById("next");
+const carouselItemsArr = Array.from(
+  document.querySelectorAll(".product-card-carousel")
+);
+const cardsContainer = document.querySelector(".cards-container");
+const carouselNavButtons = document.querySelectorAll(".carousel-nav-button");
+
+//Numbers below are used to track translate state of cards-container and to track which carousel item is currently shown
+let translateX = 0;
+let currentlyActiveIndex = 0;
+
+//
+nextButton.addEventListener("click", () => {
+  const currentlyActiveElement = document.querySelector(".active");
+  const currentlyActiveButton = document.querySelector(".button-active");
+
+  if (currentlyActiveIndex !== carouselItemsArr.length - 1) {
+    //subtract as the current item has to move to the left once next is clicked
+    translateX -= currentlyActiveElement.offsetWidth;
+    cardsContainer.style.transform = `translateX(${translateX}px)`;
+    currentlyActiveElement.classList.remove("active");
+
+    //activate next item in carousel
+    document
+      .querySelectorAll(".product-card-carousel")
+      [currentlyActiveIndex + 1].classList.add("active");
+
+    //remove active class from previously active item
+    currentlyActiveButton.classList.remove("button-active");
+
+    //accordingly update slide-buttons below
+    document
+      .querySelectorAll(".carousel-nav-button")
+      [currentlyActiveIndex + 1].classList.add("button-active");
+
+    currentlyActiveIndex++;
+  }
+});
+
+previousButton.addEventListener("click", () => {
+  const currentlyActiveElement = document.querySelector(".active");
```

Comment:
Most of code is shared between `next` and `previous` button listener handlers, I'd suggest looking into extracting this logic into a function.